packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -173,8 +178,20 @@ export function buildCart(engine: CommerceEngine, props: CartProps = {}): Cart {
       }
     },
 
-    purchase(_transactionId: string, _transactionRevenue: number) {
-      // TODO LENS-1498: log ec.purchase with all products in cart.
+    purchase(transactionId: string, transactionRevenue: number) {
```

Comment:
Is it worth making the arguments an object, since we might ask for more details about the transaction in the future? e.g. shipping and taxes.

Reply:
IMO, yes. The only concern I have is it will cause an error in Barca sports. Should it be identify as a breaking change?

Reply:
Yep, if we make this adjustment, it would be considered a breaking change. The window to make these freely will soon be closing but I think it's still open,

Reply:
This [commit](https://github.com/coveo/ui-kit/pull/3654/commits/caf3117d46a62caea8cda7c6e3e6550e2ea0cf60) contains this change (I'm open to keep it or revert it).